src/plugins/data_source/server/client/client_pool.ts

```diff
@@ -4,31 +4,32 @@
  */
 
 import { Client } from '@opensearch-project/opensearch';
+import { Client as LegacyClient } from 'elasticsearch';
 import LRUCache from 'lru-cache';
 import { Logger } from 'src/core/server';
 import { DataSourcePluginConfigType } from '../../config';
 
 export interface OpenSearchClientPoolSetup {
-  getClientFromPool: (id: string) => Client | undefined;
-  addClientToPool: (endpoint: string, client: Client) => void;
+  getClientFromPool: (id: string) => Client | LegacyClient | undefined;
+  addClientToPool: (endpoint: string, client: Client | LegacyClient) => void;
 }
```

Comment:
we have this pool includes a LRU of <endpoint, root_client> . since now we support Legacy client, shall we have one pool for legacy client, and another pool for new client? otherwise, it will cause runtime error when when a use case need a legacy client while getting a new client root client from the pool, and verse versa.

Reply:
There are in fact 2 pool instances being initiated in the constructor of `DataSourceService` [here](https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2204/files#diff-983f8d087ddcf116077576278861edbd6e4f2443a89afddc10ecfb8df456c766R35-R37). One only stores new client, the other one for legacy client. 

Since the only different of pool impl for legacy or new client is the "value" type of the LRU,  I didn't make a copy of `client_pool` to avoid having too much duplicate code. Instead I defined the LRU type as multiple type.

Reply:
I'd prefer to have 2 cache in the pool to avoid misuse the pool. but the current implementation makes sense to me.